dspace-api/src/main/java/org/dspace/discovery/indexobject/IndexFactoryImpl.java

```diff
@@ -74,35 +79,60 @@ public void writeDocument(Context context, T indexableObject, SolrInputDocument
      * Write the document to the index under the appropriate unique identifier.
      *
      * @param doc     the solr document to be written to the server
-     * @param streams list of bitstream content streams    DiscoverQueryBuilderTest.java:285
+     * @param streams list of bitstream content streams
      * @throws IOException A general class of exceptions produced by failed or interrupted I/O operations.
      */
     protected void writeDocument(SolrInputDocument doc, FullTextContentStreams streams)
             throws IOException, SolrServerException {
         final SolrClient solr = solrSearchCore.getSolr();
         if (solr != null) {
+            // If full text stream(s) were passed in, we'll index them as part of the SolrInputDocument
             if (streams != null && !streams.isEmpty()) {
-                ContentStreamUpdateRequest req = new ContentStreamUpdateRequest("/update/extract");
-                req.addContentStream(streams);
-
-                ModifiableSolrParams params = new ModifiableSolrParams();
+                // limit full text indexing to first 100,000 characters unless configured otherwise
+                final int charLimit = DSpaceServicesFactory.getInstance().getConfigurationService()
+                                                           .getIntProperty("discovery.solr.fulltext.charLimit",
+                                                                           100000);
+
+                // Use Tika's Text parser as the streams are always from the TEXT bundle (i.e. already extracted text)
+                // TODO: We may wish to consider using Tika to extract the text in the future.
+                TextAndCSVParser tikaParser = new TextAndCSVParser();
+                BodyContentHandler tikaHandler = new BodyContentHandler(charLimit);
+                Metadata tikaMetadata = new Metadata();
+                ParseContext tikaContext = new ParseContext();
+
+                // Use Apache Tika to parse the full text stream
+                try {
```

Comment:
Do we need a finally{} clause to close `streams`?

Reply:
Good catch...I think it is missing.  It's not `streams` we need to close, but `streams.getStream()` which creates a new `SequenceInputStream` that *does need to be closed*.  https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/discovery/FullTextContentStreams.java#L131

I had completely overlooked that, so I'm very glad you caught it!  I'll make sure this gets stored to a new variable & closed in a finally.

Reply:
Fixed now in https://github.com/DSpace/DSpace/pull/3020/commits/e5d5cc60e81cf19a9c2288ba6af3785b151cd1f9.  Thanks again @mwoodiupui !